components/datetime/src/provider/neo.rs

```diff
@@ -0,0 +1,156 @@
+#[derive(Debug, PartialEq, Clone)]
+#[cfg_attr(
+    feature = "datagen",
+    derive(serde::Serialize, databake::Bake),
+    databake(path = icu_datetime::provider::neo),
+)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize))]
+#[yoke(prove_covariance_manually)]
+pub enum MonthSymbolsV1<'data> {
+    /// Month codes M01, M02, M03, .. (can allow for M13 onwards)
+    ///
+    /// Found for solar and pure lunar calendars
+    Numeric(#[cfg_attr(feature = "serde", serde(borrow))] VarZeroVec<'data, str>),
+    /// Month code map that can handle arbitrary month codes including leap months
+    ///
+    /// Found for lunisolar and lunisidereal calendars
+    Map(#[cfg_attr(feature = "serde", serde(borrow))] ZeroMap<'data, UnvalidatedStr, str>),
+}
+
+
+/// Symbols used for representing the weekday name
+///
+/// This uses an auxiliary key for length.
+
+#[icu_provider::data_struct(marker(WeekdaySymbolsV1Marker, "datetime/symbols/weekdays@1"))]
+#[derive(Debug, PartialEq, Clone)]
+#[cfg_attr(
+    feature = "datagen",
+    derive(serde::Serialize, databake::Bake),
+    databake(path = icu_datetime::provider::neo),
+)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize))]
+#[yoke(prove_covariance_manually)]
+pub struct WeekdaySymbolsV1<'data> {
+    #[cfg_attr(feature = "serde", serde(borrow))]
+    symbols: VarZeroVec<'data, str>,
+}
+
+/// Symbols used for representing the day period
+///
+/// This uses an auxiliary key for length.
+
+#[icu_provider::data_struct(marker(DayPeriodSymbolsV1Marker, "datetime/symbols/dayperiods@1"))]
+#[derive(Debug, PartialEq, Clone)]
+#[cfg_attr(
+    feature = "datagen",
+    derive(serde::Serialize, databake::Bake),
+    databake(path = icu_datetime::provider::neo),
+)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize))]
+#[yoke(prove_covariance_manually)]
+pub struct DayPeriodSymbolsV1<'data> {
+    /// Stored in the order
+    /// am, pm, (noon), (midnight)
+    /// where parentheses are optional
+    #[cfg_attr(feature = "serde", serde(borrow))]
+    symbols: VarZeroVec<'data, str>,
+}
+
+/// Symbols used for representing the day name for calendars with named days (Chinese, Korean, Hindu)
+///
+/// This uses an auxiliary key for length.
+#[icu_provider::data_struct(marker(DaySymbolsV1Marker, "datetime/symbols/days@1"))]
+#[derive(Debug, PartialEq, Clone)]
+#[cfg_attr(
+    feature = "datagen",
+    derive(serde::Serialize, databake::Bake),
+    databake(path = icu_datetime::provider::neo),
+)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize))]
+#[yoke(prove_covariance_manually)]
+pub struct DaySymbolsV1<'data> {
+    #[cfg_attr(feature = "serde", serde(borrow))]
+    symbols: VarZeroVec<'data, str>,
+}
+
+/// The default per-length patterns associated with dates and times
+///
+/// This uses an auxiliary key for length. time@1 additionally uses
+/// the auxiliary key for representing hour cycle preferences.
```

Comment:
this may be annoying, bear in mind that "length" is itself two-dimensional (short/narrow/long/abbr × format/standalone), so this gets us a three-dimensional aux key total, which can make the aux key fallbacking annoying.

Fortunately we know format vs standalone and which hour cycle we need before we load the data.

Reply:
Are there really separate patterns for `format` vs `standalone`? I think that's a symbol discriminator, not a pattern discriminator.

Reply:
Oh, no, not for lengths, good point.